design/button.behavior.md

```diff
@@ -0,0 +1,70 @@
+of the button will be passed along as additional data to the app for consumption.
+
+Button Behavior Actions:
+- `hover` - The action that occurs when the button is targeted by the host app input system. 
+    - Hover states: `enter`, `hovering`, `exit`
+- `click` - The action that occurs when a complete click has happened on the button consisting of both a button pressed and button released event. 
+    - Click states: None.  This is a single atomic event.
+- `button` - The action that occurs when the button interaction state has changed from being pressed and then released.
+    - Button states: `pressed`, `holding`, `released`
+    
+Button Event Data:
+- `appSpaceTargetPoints` - The collection of target points for the current state event for the action.
+
+Button Event Data Per Action:
+- Hover Action:
+	- `enter` - single app coordinate target point for where the hover is first entered.
+	- `hovering` - collection of app coordinate target points for the time while hovering is occuring, sent at a set frequency.
+	- `exit` - single app coordinate target point for where the hover was exited.
+- Click Action:
+	- There is only the click state and will contain the single app coordinate target point of the click action.
+- Button Action:
+	- `pressed` - single app coordinate target point for where the pressed event began.
+	- `holding` - collection of app coordinate target points for the time while the button is held down, sent at a set frequency.
+	- `released` - single app coordinate target point for where the button is released.
+	
+## Architecture
+
+### ButtonEventData
+``` ts
+interface ButtonEventData {
+	appSpaceTargetPoints: TransformLike[];
+}
+```
+
+### Button Behavior Handler API
+``` ts
+* Add a hover handler to be called when the given hover state is triggered.
+ * @param hoverState The hover state to fire the handler on.
+ * @param handler The handler to call when the hover state is triggered.
+ * @return This button behavior.
+ */
+public onHover(hoverState: 'enter' | 'hovering' | 'exit', handler: ActionHandler<ButtonEventData>): this;
+    
+/**
+ * Add a click handler to be called when the given click state is triggered.
+ * @param handler The handler to call when the click state is triggered.
+ * @return This button behavior.
+ */
+public onClick(handler: ActionHandler<ButtonEventData>): this;
+
+/**
+ * Add a button handler to be called when a complete button click has occured.
+ * @param buttonState The button state to fire the handler on.
+ * @param handler The handler to call when the click state is triggered.
+ * @return This button behavior.
+ */
+public onButton(buttonState: 'pressed' | 'holding' | 'released', handler: ActionHandler<ButtonEventData>: this;
+```
+
+## Network
+
+The button behavior does not require any additional networking considerations as it will use the existing network messages within the behavior system.
```

Comment:
Would be good to document the network architecture here, even if it's not changing.

Reply:
The network architecture is specific to the behavior system.  There is no architecture for individual behaviors.  As I work on some upcoming improvements to the behavior system all up, they will be documented in there.